lms/services/email_preferences.py

```diff
@@ -112,10 +142,23 @@ def _url(self, route, h_userid, tag):
             _query={"token": self._encode_token(TokenPayload(h_userid, tag))},
         )
 
-    def instructor_digest_unsubscribe(self, h_userid) -> None:
-        """Unsubscribe `h_userid` from emails of type `tag`."""
+    def unsubscribe(self, h_userid, tag) -> None:
+        """Unsubscribe `h_userid` from instructor digest emails."""
         email_preferences = self.get_preferences(h_userid)
-        email_preferences.unsubscribe_instructor_digest()
+        if tag == EmailTypes.INSTRUCTOR_DIGEST:
+            email_preferences.mon = False
+            email_preferences.tue = False
+            email_preferences.wed = False
+            email_preferences.thu = False
+            email_preferences.fri = False
+            email_preferences.sat = False
+            email_preferences.sun = False
+        elif tag == EmailTypes.MENTION:
+            email_preferences.mention_email_subscribed = False
+        else:  # pragma: no cover
+            error_message = f"Unrecognized email tag: {tag}"
+            raise ValueError(error_message)
```

Comment:
So, if I'm getting this right, we are supposed to use the same endpoint to save email preferences, whether email digest or mentions, but not both at once, right?

And we have to send this "tag" field as discriminator to indicate which one is getting saved.

Am I understanding it?

Reply:
Ah, no, I see this is the branch for the unsubscribe link, not for the settings form. My mistake.

Reply:
Yes, this is only for the auto generated links to unsbubscribe.